### PR TITLE
chore: address broken featured collection links

### PR DESCRIPTION
**Pull request by kendallstrautman**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-kshandle-non-beta-collection-links-hashicorp.vercel.app/vault/tutorials/cross-products/kubernetes-consul-vault-pipeline#next-steps) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1201991101510148) 🎟️

## What

This PR makes a small workaround so that featured collections in tutorial views don't link to 404 pages. If the product referenced in the collection is not a part of the beta rollout, the featured collection card links externally to Learn. 

This isn't ideal UX, but I think it's _ok_ for the sake of beta. These cards linked to a mix of external and internal routes is a bit odd, but since its not a long term solution I don't think we necessarily need to adjust the cards to support true external linking. Open to feedback on that though. 

Another option woulda be to not render cards for collections that aren't included in dev dot yet 🤷 

## Testing

- [Visit this tutorial,](https://dev-portal-git-kshandle-non-beta-collection-links-hashicorp.vercel.app/vault/tutorials/cross-products/kubernetes-consul-vault-pipeline#next-steps) go to the bottom of the page. Click on the first collection card to validate that internal links still work. Then click on a collection card that says 'terraform logo' and it should link to the proper collection in production Learn.

### CODE REVIEW

File: src/views/collection-view/helpers.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { splitProductFromFilename } from 'views/tutorial-view/utils'
  * reference a different product context
  */
 
+const BETA_PRODUCTS = ['vault', 'waypoint']
```
Comment (kendallstrautman) [inline src/views/collection-view/helpers.ts:12]: Perhaps this is something we lift up, presuming other areas of the codebase may need this conditional treatment of products. Or does this already exist somewhere??

Comment (ashleemboyer) [reply src/views/collection-view/helpers.ts:12]: Agree that this should probably be higher up and I don't believe it exists yet, I'd like to try handling these with feature flags and _potentially_ adding something like `isBetaProduct` to the `CurrentProduct` context. I plan on toying around with this a little in primary nav work! What do you think of these ideas? At the very least, we can do the latter without feature flags and just hardcode the array of beta products in `CurrentProduct` context.

Comment (kendallstrautman) [reply src/views/collection-view/helpers.ts:12]: I think that's a good idea! Although in this case the collection links wouldn't be referencing the `CurrentProduct`....like they could be called within the vault product context (if that were current) and be referencing _other_ product collections, if that makes sense.

Comment (ashleemboyer) [reply src/views/collection-view/helpers.ts:12]: Ah that's a great point! I keep forgetting how that's possible in Learn content. 😅

We can probably create a new hook/util instead then that accepts a product slug as an argument, and checks against an array of beta products (like you have here) or another option is that we use a beta feature flag for each product. I think either would work! The latter would require a config change as more products are added, the former would require code changes. Both require rebuilds I believe, so I think they have the same tradeoffs!

File: src/views/collection-view/helpers.ts [hunk 2]
```diff
@@ -20,5 +20,14 @@ export function getTutorialSlug(
 
 export function getCollectionSlug(collectionDbSlug: string): string {
   const [product, collectionFilename] = collectionDbSlug.split('/')
+  const isBetaProduct =
+    __config.dev_dot.products_with_content_preview_branch.includes(product)
+
+  // if not a 'sanctioned product', link externally to Learn
+  // interim solution for BETA where not all products are onboarded
+  if (isBetaProduct) {
```
Comment (ashleemboyer) [inline src/views/collection-view/helpers.ts:28]: I think this needs flopped!

```suggestion
  if (!isBetaProduct) {
```

Comment (ashleemboyer) [reply src/views/collection-view/helpers.ts:28]: Wait... I think I mean flipped? Words are hard. Flooped? 🤣

Comment (kendallstrautman) [reply src/views/collection-view/helpers.ts]: 🤣

**Review by ashleemboyer — APPROVED** [review]
Works beautifully! ✨

**Comment by kendallstrautman** [comment]
@ashleemboyer I followed up and removed that duplicate `getCollectionSlug` function as well 😄

**Pull request merged**